data/core/object.lua

```diff
@@ -1,12 +1,20 @@
+---Base class providing OOP functionality for Lua.
+---All classes in Pragtical inherit from Object.
 ---@class core.object
 ---@field super core.object
 local Object = {}
```

Comment:
I forgot about this one, an overload that should be added to all classes:

```lua
---@overload fun():core.object
```
This helps luals know that a call like `local var = Object()` would mean that var is a now an instance of core.object